Backend/Backend.Tests/SubscriptionTest.cs

```diff
@@ -0,0 +1,247 @@
+            _httpContextMock = new();
+            _subscriptionController = new(_authServiceMock.Object, _subscriptionService)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = _httpContextMock.Object
+                }
+            };
+
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+        }
+      
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        public Task InitializeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            // Re-initilize every test
+            _dbContextMock = new(new DbContextOptions<AppDbContext>());
+            _authServiceMock = new();
+            _profileServiceMock = new();
+            _subscriptionService = new(_dbContextMock.Object);
+            _httpContextMock = new();
+            _subscriptionController = new( _authServiceMock.Object,_subscriptionService)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = _httpContextMock.Object
+                }
+            };
+
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+        }
+
+        [Fact]
+        public void Subscribe_Podcast_Success()
+        {
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+            bool response = false;
+
+            try
+            {
+                response = _subscriptionService.SubscribeAsync(podcast.Object.First().Id, user.Object.Skip(1).First()).Result;
+
+            }
+            catch (Exception e)
+            {
+                Assert.Fail(e.Message);
+
+            }
+
+            Assert.True(response);
+
+        }
+        [Fact]
+        public void unsubscribe_Podcast()
+        {
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+            bool response = false;
+
+            try
+            {
+                response = _subscriptionService.UnsubscribeAsync(podcast.Object.First().Id, user.Object.First()).Result;
+
+            }
+            catch (Exception e)
+            {
+                Assert.Fail(e.Message);
+
+            }
+            Assert.True(response);
+        }
+
+        [Fact]
+        public void issubscribed_Podcast()
+        {
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+            bool response = false;
+
+            try
+            {
+                response = _subscriptionService.IsSubscribed(podcast.Object.First().Id, user.Object.First()).Result;
+
+            }
+            catch (Exception e)
+            {
+                Assert.Fail(e.Message);
+
+            }
+            Assert.True(response);
+        }
+
+        //[Fact]
```

Comment:
Test is commented out

Reply:
Dont close one of these convos, cus I need to add the tests still and closing the convos will let the merge happen

Reply:
redid all tests. Should be fine